csharp/README.md

```diff
@@ -0,0 +1,34 @@
+# Prerequisites
+
+- jose-jwt
+- commandlineparser
+
+## Instructions
+
+First install project dependencies:
+
+```bash
+dotnet restore
+```
+
+## Example
+
+This assumes you have generated a key pair using the `generate_key_pair.sh` script in the root folder of the project.
+
+```bash
+# Assumes you are running from within the `csharp` directory
+dotnet run --project src/signing.csproj -- --kid $kid --key-file ./ec512-private-key.pem --payload ./payload.json
+
+# slurp into shell variable
+signature=$(dotnet run --project src/signing.csproj -- --kid $kid --key-file ./ec512-private-key.pem --payload ./payload.json 2>/dev/null)
```

Comment:
From `src/signing.csproj` I assume you're meant to run these from the _csharp_ dir.

```suggestion
dotnet run --project src/signing.csproj -- --kid $kid --key-file ../ec512-private-key.pem --payload ../payload.json

# slurp into shell variable
signature=$(dotnet run --project src/signing.csproj -- --kid $kid --key-file ../ec512-private-key.pem --payload ../payload.json 2>/dev/null)
```

Reply:
I actually ran these from within the csharp directory and copied my assets there. Taking from the parent is probably more consistent though :+1: